status-reporter/status_reporter/status_reporter.py

```diff
@@ -0,0 +1,140 @@
+                    self._reporter_token,
+                    self._reporter_user_name
+                    ]):
+            invalid_conf_settings = {
+                'Managers Ips': self._managers_ips,
+                'Username': self._reporter_user_name,
+                'Password token': self._reporter_token
+            }
+            raise InitializationError('Please verify the reporter\'s config related to '
+                                      '{0} ..'.format(json.dumps(invalid_conf_settings,
+                                                                 indent=1)))
+
+        if self._ca_path and os.path.exists(self._ca_path):
+            raise InitializationError('CA certificate was not found, '
+                                      'please verify the given location %s', self._ca_path)
+
+        self._current_reporting_freq = self._config.get('reporting_freq', None)
+        self._component_type = component_type
+
+    def _build_auth_headers(self):
+        return {
+            'Authorization': 'Basic ' + base64.b64encode('{0}:{1}'.format(
+                self._reporter_user_name,
+                self._reporter_token)
+            ),
+            'tenant': 'default_tenant'
+        }
+
+    def _build_full_reporting_url(self, ip):
+        # TODO: UUID still needs to be added here
+        return 'https://{0}/cluster_status/{1}'.format(ip, self._component_type)
+
+    def _build_report(self, status):
+        if 'reporting_freq' in status:
+            logger.error('The status report contained reserved keywords,'
+                         ' please verify the report content %s', status)
+            return None
+
+        status['reporting_freq'] = self._current_reporting_freq
+        return status
+
+    def _report(self):
+        status = self.status_sampler()
+        if not isinstance(status, dict):
+            logger.error('Received status report not in required'
+                         ' "dict" type %s, throwing the status '
+                         'away...', status)
+            return
+
+        report = self._build_report(status)
+        if report is None:
+            return
+
+        # If there is a malfunctioning manager, let's try to avoid using the same
+        # manager always.
+        random.shuffle(self._managers_ips)
+
+        # TODO: read the managers' ip from the rest endpoint, so we should not have to update this
+        for manager_ip in self._managers_ips:
+            try:
+                requests.put(self._build_full_reporting_url(manager_ip),
+                             data=report,
+                             headers=self._build_auth_headers(),
+                             verify=self._ca_path)
+                return
+            except Exception as e:
+                logger.debug('Error had occurred while trying to report '
+                             'status: ' + str(e))
+        logger.error('Could not find active manager to report the current status,'
```

Comment:
i'd also mention the manager here, eg. `'Could not find an active manager to report the current status, tried %s', ','.join(self._managers_ips)`